modules/social_features/social_embed/social_embed.module

```diff
@@ -41,3 +42,113 @@ function social_embed_create_media_button() {
   $button->save();
 
 }
+
+/**
+ * Helper function for granting permissions.
+ */
+function _social_embed_create_permissions() {
+  user_role_grant_permissions('sitemanager', ['administer social embed settings']);
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function social_embed_form_user_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  /** @var \Drupal\Core\Entity\ContentEntityFormInterface $form_object */
+  $form_object = $form_state->getFormObject();
+
+  /** @var \Drupal\social_user\Entity\User $account */
+  $account = $form_object->getEntity();
+
+  /** @var \Drupal\user\UserDataInterface $user_data */
+  $user_data = \Drupal::service('user.data');
+
+  $user_states = $user_data->get('social_embed', $account->id(), 'consent');
+
+  $form['embed_consent'] = [
+    '#type' => 'fieldset',
+    '#title' => [
+      'text' => [
+        '#markup' => t('<strong>Embed consent</strong>'),
+      ],
+      'icon' => [
+        '#markup' => '<svg class="icon icon-expand_more"><use xlink:href="#icon-expand_more" /></svg>',
+        '#allowed_tags' => ['svg', 'use'],
+      ],
+    ],
+    '#tree' => TRUE,
+    '#collapsible' => TRUE,
+    '#collapsed' => TRUE,
+    '#attributes' => [
+      'class' => [
+        'form-horizontal',
+        'form-embed-consent',
+      ],
+    ],
+  ];
+
+  $form['embed_consent']['description'] = [
+    '#type' => 'html_tag',
+    '#tag' => 'p',
+    '#value' => t('The following options enables you to choose which embed you would allow in your stream. For example, YouTube videos.'),
+  ];
+
+  $providers = [
```

Comment:
Let's not make a list like this, rather invoke the whitelisted list so we don't have to keep it up to date in two places.

Reply:
(Also that is out of scope for this story so don't worry)